### PR TITLE
adding datatree parameters

### LINKED ISSUE #164: AttributeError: 'DataTree' object has no attribute 'sweep_group_name'
* xradar version:  0.4.3
* Python version: 3.12
* Operating System: Ubuntu

### Description

Hi every one, 

I've tried to reproduce this [grinding](https://arm-doe.github.io/pyart/examples/xradar/plot_grid_xradar.html) example using sigmet files available in amazon. 

```python 
import xradar as xd
import fsspec
from datetime import datetime
import pyart
import matplotlib.pyplot as plt


def create_query(date, radar_site):
    """
    Creates a string for quering the IDEAM radar files stored in AWS bucket
    :param date: date to be queried. e.g datetime(2021, 10, 3, 12). Datetime python object
    :param radar_site: radar site e.g. Guaviare
    :return: string with a IDEAM radar bucket format
    """
    if (date.hour != 0) and (date.minute != 0):
        return f"l2_data/{date:%Y}/{date:%m}/{date:%d}/{radar_site}/{radar_site[:3].upper()}{date:%y%m%d%H%M}"
    elif (date.hour != 0) and (date.minute == 0):
        return f"l2_data/{date:%Y}/{date:%m}/{date:%d}/{radar_site}/{radar_site[:3].upper()}{date:%y%m%d%H}"
    else:
        return f"l2_data/{date:%Y}/{date:%m}/{date:%d}/{radar_site}/{radar_site[:3].upper()}{date:%y%m%d}"


def main():
    date_query = datetime(2022, 10, 6)
    radar_name = "Guaviare"
    query = create_query(date=date_query, radar_site=radar_name)
    str_bucket = "s3://s3-radaresideam/"
    fs = fsspec.filesystem("s3", anon=True)
    radar_files = sorted(fs.glob(f"{str_bucket}{query}*"))
    file = fsspec.open_local(
        f"filecache::s3://{radar_files[0]}",
        s3={"anon": True},
        filecache={"cache_storage": "/tmp/radar"},
    )

    tree1 = xd.io.open_iris_datatree(file)
    radar = pyart.xradar.Xradar(tree1, default_sweep='sweep_0')
    # Grid using 11 vertical levels, and 101 horizontal grid cells at a resolution on 1 km
    grid = pyart.map.grid_from_radars(
        (radar,),
        grid_shape=(1, 501, 501),
        grid_limits=(
            (0.0, 10_000),
            (-300_000.0, 300_000.0),
            (-300_000, 300_000.0),
        ),
    )

    display = pyart.graph.GridMapDisplay(grid)
    display.plot_grid(
        "DBZH", level=0, vmin=-20, vmax=60, cmap="pyart_ChaseSpectral"
    )
    plt.show()
    print(1)


if __name__ == "__main__":
    main()
```
However, I got the following error, 

```python 
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[6], line 1
----> 1 radar = pyart.xradar.Xradar(tree1, default_sweep='sweep_0')

File [~/miniconda3/envs/raw2zarr/lib/python3.12/site-packages/pyart/xradar/accessor.py:274](http://127.0.0.1:7022/~/miniconda3/envs/raw2zarr/lib/python3.12/site-packages/pyart/xradar/accessor.py#line=273), in Xradar.__init__(self, xradar, default_sweep, scan_type)
    272 self.xradar = xradar
    273 self.scan_type = scan_type or "ppi"
--> 274 self.combined_sweeps = self._combine_sweeps(self.xradar)
    275 self.fields = self._find_fields(self.combined_sweeps)
    276 self.time = dict(
    277     data=(self.combined_sweeps.time - self.combined_sweeps.time.min()).astype(
    278         "int64"
   (...)
    282     calendar="gregorian",
    283 )

File [~/miniconda3/envs/raw2zarr/lib/python3.12/site-packages/pyart/xradar/accessor.py:552](http://127.0.0.1:7022/~/miniconda3/envs/raw2zarr/lib/python3.12/site-packages/pyart/xradar/accessor.py#line=551), in Xradar._combine_sweeps(self, radar)
    549 def _combine_sweeps(self, radar):
    550     # Loop through and extract the different datasets
    551     ds_list = []
--> 552     for sweep in radar.sweep_group_name.values:
    553         ds_list.append(radar[sweep].ds.drop_duplicates("azimuth"))
    555     # Merge based on the sweep number

File [~/miniconda3/envs/raw2zarr/lib/python3.12/site-packages/datatree/common.py:54](http://127.0.0.1:7022/~/miniconda3/envs/raw2zarr/lib/python3.12/site-packages/datatree/common.py#line=53), in TreeAttrAccessMixin.__getattr__(self, name)
     52         with suppress(KeyError):
     53             return source[name]
---> 54 raise AttributeError(
     55     f"{type(self).__name__!r} object has no attribute {name!r}"
     56 )

AttributeError: 'DataTree' object has no attribute 'sweep_group_name'
```
Digging around, I found out that the `xd.io.open_iris_datatree` data loader is not adding the `sweep_group_name`, `sweep_fixed_angle`, and `radar_parameters` to the datatree. I worked around and I submitted a push request where this issue is solved following the FM301 standard WMO data model

### PR DESCRIPTION
**Pull request by aladinor**
adding datatree parameters sweep_group_name, sweep_fixed_angle, and radar_parameters to the iris datatree following the FM301 standard WMO data model

- [ ] Closes #164

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@aladinor Can you rebase on latest main? Your changes are currently on top of xradar 0.2.0.

File: xradar/io/backends/iris.py [hunk 1]
```diff
@@ -3917,6 +3922,72 @@ def _get_iris_group_names(filename):
...
+    # keep only defined mandatory and defined optional attributes per default
+    attrs = root.attrs.keys()
+    remove_attrs = set(attrs) ^ set(required_global_attrs)
+    if optional:
+        remove_attrs ^= set(optional_root_attrs)
+    for k in remove_attrs:
+        root.attrs.pop(k, None)
+
+    # handle sweep variables and dimension
+    root = root.rename_vars({"sweep_number": "sweep_group_name"})
```
Comment (kmuehlbauer) [inline xradar/io/backends/iris.py:3962]: @aladinor Please see further discussion in #164. We should remove `sweep_group_name` and `sweep_fixed_angle` from xradar data model.

**Comment by mgrover1** [comment]
@aladinor - I can open an issue on the Py-ART side making those fields not required.

File: xradar/io/backends/iris.py (hunk 1 above)
Comment (aladinor) [reply xradar/io/backends/iris.py:3962]: Should I remove it from model.py file ?
https://github.com/openradar/xradar/blob/0145954f3847abd89208c46055241aeae51316c8/xradar/model.py#L112C1-L121C2

Comment (kmuehlbauer) [reply xradar/io/backends/iris.py:3962]: Yes, but this might show usages in other parts of xradar. If you need any help cleaning up the mess, let us know.

**Comment by aladinor** [comment]
I will open a new PR since I was working on an outdated folk and I need to work on the required root variables.

**Comment by kmuehlbauer** [comment]
@aladinor You could just force-push the needed changes to your branch. No need to open a new PR. But do as you feel most comfortable with.

**Comment by aladinor** [comment]
I was worried about messing everything up. I just decided to start over.

**Pull request closed without merging**